### DISCUSSION TITLE
Conventional way of mapping backend errors to fields in form submit?

### DISCUSSION THREAD

**Discussion by bombillazo** [ant-design/ant-design#42842] [Q&A] [answered]
Hello,

Sometimes even when validation passes in the front-end, things can go wrong in the backend due to an invalid value of a field, especially when it comes to database errors. What is the AntD common way to respond to these errors and allow users to correct the form values given the corresponding error received from the backend?

**Comment by kiner-tang** [comment, marked as answer]
If the backend validation results are incorrect, we usually send a `message.error` to warning user to change the form value manually, but if you want the backend to correct the form value when it returns an error, you can also try using `form.setFieldsValue({...})`
